chat-server/server.js

```diff
@@ -21,6 +23,38 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT,() => {
+let nextMessageId = 1;
+
+app.post("/messages", function (request, response) {
+  const newMessage = {
+    id: nextMessageId++,
+    from: request.body.from,
+    text: request.body.text,
+  };
+
+  messages.push(newMessage);
+  response.status(201).json(newMessage);
+});
+
+app.get("/messages", function (request, response) {
+  response.json(messages);
+});
+
+app.get("/messages/:id", function (request, response) {
+  const messageId = parseInt(request.params.id);
+  let message = null;
+  messages.forEach((singleMsg) => {
+    if (singleMsg.id === messageId) {
+      message = singleMsg;
+    }
+  });
```

Comment:
What do you think of replacing this logic with messages.find()? It'll be easier to read and more concise.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Thank you, @sherif98 . I've updated my code. I've never used the .find method before, but I agree that it makes the code easier to read.